analyses/hotspots-detection/01-create-hotspot-maf.Rmd

```diff
@@ -0,0 +1,246 @@
+---
+title: "Create consensus hotspot maf"
+author: Krutika Gaonkar for D3b
+output: html_notebook
+params:
+  vaf_filter:
+    label: VAF filter for hotspots variant calls
+    value: 0
+    input: integer  
+---
+
+In the previous step we gather all known cancer hotspots by scavenging calls by
+- checking for overlap with amino acid positions in a curated and published cancer hotspot [database](https://www.cancerhotspots.org/files/hotspots_v2.xls)
+- checking for overlap with non-coding region hotspots mutations , here we are using the in TERT promoter region
+
+In this notebook, we will create consensus mafs from the filtered call `scratch/hotspot-detection/*RDS`. Since there are slight differences in read support from each caller, we will take a mean value to provide the information per site in the consensus maf file. All other vcf2maf columns will be added from strelka/mutect2  
+
+## Setup
+```{r}
+library("tidyverse")
+library("maftools")
+
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+
+results_dir <- file.path(root_dir,
+                         "analyses",
+                         "hotspots-detection",
+                         "results")
+# Make output folder
+if (!dir.exists(results_dir )) {
+  dir.create(results_dir , recursive = TRUE)
+}
+
+scratch_dir <- file.path(
+  root_dir,
+  "scratch",
+  "hotspot-detection"
+)
+
+# Import function to split MNV to SNV
+source(file.path(root_dir, "analyses", "snv-callers", "util", "split_mnv.R"))
+
+maf_coltypes <- readRDS(file.path("input", "maf_coltypes.RDS"))
```

Comment:
A heads up: with the changes in MAF files coming with normalization (https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/990), the set of column headers is changing, so this data file will need to be regenerated as some of the columns are missing. 

Note also that there are unique columns for each caller (see https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/990#issuecomment-826879406) that may be best filtered out in `00-subset-maf.R`

Reply:
This is updated as of #1050